### DISCUSSION TITLE
将标注以 MD 2.0 模板形式导出

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4321] [插件说明]
描述
---

> 简悦可以将当前页面的标注导出为 Markdown 但仅支持 Markdown 定制化 1.0。如果希望使用 [Markdown 定制化 2.0](https://github.com/Kenshin/simpread/discussions/3725#discussioncomment-2635600) 的话，则可使用此插件。

地址
---

[https://simpread.ksria.cn/plugins/details/8jnsO2OtHW](https://simpread.ksria.cn/plugins/details/8jnsO2OtHW)

快捷键
---

全局快捷键 `m t`

权限
---

因为涉及到 **[Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725)**，所以此插件为 **高级账户权限** 才能使用。

视频
---

https://user-images.githubusercontent.com/81074/182790631-61aaf6b7-76c9-42f9-bcfe-99d615616166.mp4

什么时候使用
---

- 如果你在使用 Logseq 并且是 Markdown 格式的时候

- 需要使用 Markdown 模板 2.0 的时候

自动化
---

1. 利用 [自动化辅助增强](https://github.com/Kenshin/simpread/discussions/3596) + [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725) + [此模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-3255872) ，可以实现：当稍后读、标注变化时自动生成标注文件

2. 利用同步助手 的导出增强，可以将 `.md` 导出到任意位置，如设置为 Logseq 的 pages 文件夹，进而实现完美的：在网页标注，自动同步到 Logseq 的功能。

上述配置 [请看这里](https://github.com/Kenshin/simpread/discussions/4321#discussioncomment-3321977)。

选项
---

> 随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角触发器 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

![image](https://user-images.githubusercontent.com/81074/182791281-579fb126-d265-4433-9ad0-b714a631ca3c.png)

功能说明
---

1. 定制导出文件名

   支持占位符，详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/4321#discussioncomment-3321975)。

   ![image](https://user-images.githubusercontent.com/81074/165071299-9713af43-a706-47ed-a7e8-8be05ed4a2a2.png)

2. 仅支持同步助手导出方案

3. 支持手动导出或 [自动化导出](https://github.com/Kenshin/simpread/discussions/4321#discussioncomment-3321977) 方案
    
4. 目前只能开启 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725) 后才能使用此功能。

    ![image](https://user-images.githubusercontent.com/81074/165073027-011b550b-dcf6-44ee-923a-5680547894a0.png)

引申
---

1. 此插件不仅适用于 Logseq，只要是希望通过 [Markdown 2.0 模板](https://github.com/Kenshin/simpread/discussions/3725) 导出 `.md` 均可使用此插件，如果你希望使用 Orgmode 格式的话，可以看 [导出 org-mode](https://github.com/Kenshin/simpread/discussions/3775)。

2. 如果不使用 Markdown 2.0 模板的话，可使用简悦内置方案。

3. 如果是 Obsidain 用户的话，可以使用 [导入到 Obsidian](https://github.com/Kenshin/simpread/discussions/2902) 实现更多 Obsidian 相关功能。

4. 此插件也可以配合 [Logseq SimpRead Sync](https://github.com/Kenshin/simpread/discussions/4117) 实现更多用法。

更新日志
---

### 1.0.1

- `Fix` 当标题出现特殊字符时无法使用的错误。

### 1.0.0

- 手动导出，支持快捷键 `m t`

- 自动化导出

- 选项

    - 自定义标题

    - 使用 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725)

- 权限验证

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by Kenshin** [comment]
# 自定义标题

一般情况下不用修改自定义标题的默认值，因为默认值跟 [知识库](https://kb.simpread.pro/#/page/%E5%BB%BA%E7%AB%8B%E7%9F%A5%E8%AF%86%E5%BA%93) 是一个命名体系。

| 占位符             | 描述                                                                                     |
|-----------------|----------------------------------------------------------------------------------------|
| `{{id}}`        | 稍后读 ID                                                                                 |
| `{{title}}`     | 稍后读的标题                                                                                 |
| `{{un_title}}`  | 稍后读的标题                                                                                 |
| `{{title}}`     | 原页面的标题                                                                                 |
| `{{timestamp}}` | 当前稍后读建立时的时间戳 e.g. 20212809162353                                                       |
| `{{now\|<格式化>}}` | [详细说明](https://github.com/Kenshin/simpread/discussions/3725#discussioncomment-2577619) |
| `{{note}}`      | 当前稍后读的备注，当不存在时使用稍后读标题，此功能可以理解为别名。                                                                               |
| `{{mode}}`     | 默认值为 `@annote`，一般不用设定                                                                                 |

**Comment by Kenshin** [comment]
# 自动化方案

与 [导出到 Obsidian](https://github.com/Kenshin/simpread/discussions/2902) 插件方案一致。

```
m d u n
m d u u
m d a a
m d a c
```

将上面的自动化快捷键依次加入到 [自动化辅助增强](https://github.com/Kenshin/simpread/discussions/3596) 插件，如下图所示

![image](https://user-images.githubusercontent.com/81074/182777228-f81da5a1-e614-4b51-93f2-3e76fcf71596.png)

## 模板

安装 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725)，进入 [Markdown 2.0 模板库](https://github.com/Kenshin/simpread/discussions/3725#discussioncomment-2635600) 随便找一个关于 Logseq 的模板，例如 [此模板](https://github.com/Kenshin/simpread/discussions/2153#discussioncomment-3255872) 复制到 Markdown 辅助增强插件。

![image](https://user-images.githubusercontent.com/81074/182777826-cb86026a-0a17-420c-a4b3-b3370d343313.png)

需要开启下图所示的选项才可使用

![image](https://user-images.githubusercontent.com/81074/165034089-a8e2987e-7947-4bdc-89e4-5d743fff1f47.png)

## 同步助手的增强导出

同步助手的增强导出方案可以将任意指定的文件转移到特定目录，如果是 Logseq 的话，可以使用下面的方案

```
{"extension":"md", "path":"C:\\xxx\\xxx\\xxx\\Documents\\Logseq\\pages"}
```

这样当生成 `.md` 时会自动导出到 Logseq 下面的 pages 文件夹进而实现自动化。

**Comment by sokouu** [comment]
多谢老大这么快就搞出这个插件, 补充自己摸索到的一点, 在高级设定-导出时可选中, 需要选择全文(含标注), 不然会产生conflict文件.

**Reply by Kenshin** [reply]
是的，所有导出相关的功能使用前提都是需要设置知识库（你提到的方式就是知识库的设置之一。）

**Reply by sokouu** [reply]
我原来是按教程“选择全文(含标注)+标注”, 发现这样会生产一个conflict的标注文件, 如果就只选全文, 只让自动化辅助增强插件来生成标注, 就没问题了.

**Reply by Kenshin** [reply]
嗨，之前说的有误，确实如此。

我调整了算法，你可以重新安装这个插件再试试，是否可以了。

测试方式：选择【全文(含标注) + 标注】，看看标注是否正确。

另外：当选择了【全文(含标注) + 标注】，其实也可以通过删除自动化设置里面的 Markdown 任务也可以解决这个问题。

**Reply by sokouu** [reply]
刚做了测试,这回不会出现冲突文件, 但是生成的标注还是md1.0的样式, 回归到"全文(含标注)", 利用自动化任务就没问题.

**Reply by Kenshin** [reply]
试试：删除插件再重新安装。（我调整了导出顺序，应该没问题才对）
